helix-core/src/main/java/org/apache/helix/util/StatusUpdateUtil.java

```diff
@@ -560,6 +565,10 @@ void publishStatusUpdateRecord(ZNRecord record, Message message, Level level,
    */
   void publishErrorRecord(ZNRecord record, String instanceName, String updateSubPath,
       String updateKey, String sessionId, HelixDataAccessor accessor, boolean isController) {
+    _logger.error("StatusUpdate Error record: {}", record.toString());
+    if (!ERROR_LOG_TO_ZK_ENABLED) {
```

Comment:
IMHO,

if (!ERROR_LOG_TO_ZK_ENABLED) {
  _logger.error("StatusUpdate Error record: {}", record.toString());
  return;
}
...

Reply:
Not actually, even we disable writting to ZK, we still need to log into local log file, right?

Reply:
I don't have a strong preference here.
But I think it depends on how we treat this util, is it a more powerful logger? Or it is a status updater like the name suggested.

The current implementation is treating this class as a pure status updater. There is no extra log done unless the update process fails. And I think it is clean. If a log is needed, then the caller shall call logger method in parallel with the publishErrorRecord() method.

On the other hand, if we want to twist this class to be a more powerful logger, then we will have mixed usage. Only part of the code is using it. And whether to use it or not is not clear.

Overall, I treat the log here as a fallback logic when ERROR_LOG_TO_ZK_ENABLED == false. So it shall not log when the util still writes to ZK.

Reply:
So let us keep it this way. The reasoning is that source of truth logging is in log file.

Reply:
I'm fine with both designs. But as I mentioned, this change is half-way. Please either ensure that the source of truth is with the log for ALL the status util call (including log error, log status, etc.) or please don't log here otherwise the util is mixed up with some methods with log and the others don't.

Reply:
Good point. The log are in `publishStatusUpdateRecord' and `publishErrorRecord`.

Here publishStatusUpdateRecord would never log to Zk anymore. The participant side, it has trace level log to log4j, but not controller side. Let me revise it to the way that both controller and participant side has log to log4j then.